msdk-io/msdk-io-mzml2/src/main/java/io/github/msdk/io/mzml2/MzMLSpectrum.java

```diff
@@ -154,95 +148,153 @@ public Integer getNumberOfDataPoints() {
 
       compressions.add(getIntensityBinaryDataInfo().getCompressionType());
 
-      InputStream encodedIs = new ByteBufferInputStreamAdapter(mappedByteBufferInputStream, 0,
+      InputStream encodedIs = new ByteBufferInputStreamAdapter(mappedByteBufferInputStream,
+          getIntensityBinaryDataInfo().getPosition(),
           getIntensityBinaryDataInfo().getEncodedLength());
       InputStream decodedIs = Base64.getDecoder().wrap(encodedIs);
       byte[] decodedData = IOUtils.toByteArray(decodedIs);
+      System.out.println(new String(IOUtils.toByteArray(encodedIs)));
 
-      result = MzMLIntensityPeaksDecoder.decode(decodedData, 0, precision,
+      intensityValues = MzMLIntensityPeaksDecoder.decode(decodedData, decodedData.length, precision,
           getIntensityBinaryDataInfo().getArrayLength(), compressions).arr;
     } catch (Exception e) {
       throw (new MSDKRuntimeException(e));
     }
 
-    return result;
+    return intensityValues;
   }
 
   @Override
-  public Float getTIC() {
-    // TODO Auto-generated method stub
-    return null;
+  public MsSpectrumType getSpectrumType() {
```

Comment:
Actually, the mzML specification has CV terms for profile and centroid spectra. So it would be better to read the SpectrumType from the CV term, and only use the autodetection if it was not specified in the mzML file